### PR TITLE
Fixes various code style issues and integrate PHP CS Fixer into drone

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
The https://github.com/PHP-CS-Fixer/PHP-CS-Fixer project is another code style fixer for PHP which we have a config for since #10992 but actually never used. This pr integrates it into drone and fixes the code base, primarily adding a trailing comma on multiline arrays where phpcs has a [big weakness](https://github.com/squizlabs/PHP_CodeSniffer/issues/582).

On the other side the PHP CS Fixer [can't properly handle files where HTML syntax is mixed with PHP code](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/3702#issuecomment-396717120), so it makes sense to use both projects in our CI pipeline.

This pr is a result from https://github.com/joomla/joomla-cms/pull/39739#pullrequestreview-1274108418.

This can be merged by review.

### CODE REVIEW

File: administrator/templates/atum/error_login.php [hunk 1]
```diff
@@ -129,19 +129,19 @@
...
+                                        <?php // Make the first assignment to setError() outside the loop so the loop does not skip Exceptions?>
                                         <?php $this->setError($this->_error->getPrevious()); ?>
                                         <?php while ($loop === true) : ?>
                                             <p><strong><?php echo Text::_('JERROR_LAYOUT_PREVIOUS_ERROR'); ?></strong></p>
                                             <p><?php echo htmlspecialchars($this->_error->getMessage(), ENT_QUOTES, 'UTF-8'); ?></p>
                                             <?php echo $this->renderBacktrace(); ?>
                                             <?php $loop = $this->setError($this->_error->getPrevious()); ?>
                                         <?php endwhile; ?>
-                                        <?php // Reset the main error object to the base error ?>
+                                        <?php // Reset the main error object to the base error?>
```
Comment (HLeithner) [inline administrator/templates/atum/error_login.php]: That looks odd

Comment (laoneo) [reply administrator/templates/atum/error_login.php]: Comes from https://cs.symfony.com/doc/rules/comment/no_trailing_whitespace_in_comment.html

Comment (laoneo) [reply administrator/templates/atum/error_login.php]: I'm excluding also layout files now

File: administrator/components/com_config/src/Controller/ComponentController.php [hunk 2]
```diff
@@ -156,7 +156,7 @@ public function save($key = null, $urlVar = null)
             case 'save':
                 $this->setMessage(Text::_('COM_CONFIG_SAVE_SUCCESS'), 'message');
 
-                // No break
+                // no break
```
Comment (HLeithner) [inline administrator/components/com_config/src/Controller/ComponentController.php]: hmm comments get changed to lower case? this is wrong we always start uppercase in comments

File: administrator/components/com_content/src/Controller/ArticleController.php [hunk 3]
```diff
@@ -77,16 +77,16 @@ protected function postSaveHook(BaseDatabaseModel $model, $validData = [])
...
+            $editState['link']          = $link;
+            $editState['title']         = $model->getItem($id)->title;
+            $editState['type']          = $type;
             $editState['request']['id'] = $id;
 
             $this->app->setUserState('com_menus.edit.item', [
                 'data' => $editState,
                 'type' => $type,
-                'link' => $link]);
+                'link' => $link, ]);
```
Comment (HLeithner) [inline administrator/components/com_content/src/Controller/ArticleController.php]: I'm suprised that this works for phpcs

File: administrator/components/com_users/src/Model/UsersModel.php [hunk 4]
```diff
@@ -549,7 +549,7 @@ private function buildDateRange($range)
             case 'post_year':
                 $dNow = false;
 
-                // No break
+                // no break
```
Comment (HLeithner) [inline administrator/components/com_users/src/Model/UsersModel.php]: shouldn't be lowercase

File: components/com_contact/src/Controller/DisplayController.php [hunk 5]
```diff
@@ -73,9 +73,9 @@ public function display($cachable = false, $urlparams = [])
         }
 
         $safeurlparams = ['catid' => 'INT', 'id' => 'INT', 'cid' => 'ARRAY', 'year' => 'INT', 'month' => 'INT',
-            'limit' => 'UINT', 'limitstart' => 'UINT', 'showall' => 'INT', 'return' => 'BASE64', 'filter' => 'STRING',
-            'filter_order' => 'CMD', 'filter_order_Dir' => 'CMD', 'filter-search' => 'STRING', 'print' => 'BOOLEAN',
-            'lang' => 'CMD'];
+            'limit'               => 'UINT', 'limitstart' => 'UINT', 'showall' => 'INT', 'return' => 'BASE64', 'filter' => 'STRING',
+            'filter_order'        => 'CMD', 'filter_order_Dir' => 'CMD', 'filter-search' => 'STRING', 'print' => 'BOOLEAN',
+            'lang'                => 'CMD', ];
```
Comment (HLeithner) [inline components/com_contact/src/Controller/DisplayController.php]: this array is strange also the ending ] is not on a new line

File: components/com_newsfeeds/src/Controller/DisplayController.php [hunk 6]
```diff
@@ -41,12 +41,12 @@ public function display($cachable = false, $urlparams = false)
...
 
-        if ($this->app->getIdentity()->get('id') || ($this->input->getMethod() === 'POST' && $vName === 'category' )) {
+        if ($this->app->getIdentity()->get('id') || ($this->input->getMethod() === 'POST' && $vName === 'category')) {
             $cachable = false;
         }
 
-        $safeurlparams = ['id' => 'INT', 'limit' => 'UINT', 'limitstart' => 'UINT',
-                                'filter_order' => 'CMD', 'filter_order_Dir' => 'CMD', 'lang' => 'CMD'];
+        $safeurlparams = ['id'                 => 'INT', 'limit' => 'UINT', 'limitstart' => 'UINT',
+                                'filter_order' => 'CMD', 'filter_order_Dir' => 'CMD', 'lang' => 'CMD', ];
```
Comment (HLeithner) [inline components/com_newsfeeds/src/Controller/DisplayController.php]: looks odd

File: libraries/src/Form/Field/TransitionField.php [hunk 7]
```diff
@@ -162,7 +162,7 @@ function ($item) use ($user, $component) {
         $workflowName = $db->setQuery($query)->loadResult();
 
         $default = [[
-            HTMLHelper::_('select.option', '', Text::_($workflowName))]];
+            HTMLHelper::_('select.option', '', Text::_($workflowName)), ]];
```
Comment (HLeithner) [inline libraries/src/Form/Field/TransitionField.php]: multiline array seems to be a strange definition...

File: libraries/src/Form/FormField.php [hunk 8]
```diff
@@ -509,7 +509,7 @@ public function __set($name, $value)
                 // Removes spaces from left & right and extra spaces from middle
                 $value = preg_replace('/\s+/', ' ', trim((string) $value));
 
-                // No break
+                // no break
```
Comment (HLeithner) [inline libraries/src/Form/FormField.php]: uppercase

File: libraries/src/Form/FormField.php [hunk 9]
```diff
@@ -539,40 +539,40 @@ public function __set($name, $value)
...
+                $this->name      = $this->getName($this->fieldname);
                 break;
 
             case 'multiple':
                 // Allow for field classes to force the multiple values option.
                 $value = (string) $value;
                 $value = $value === '' && isset($this->forceMultiple) ? (string) $this->forceMultiple : $value;
 
-                // No break
+                // no break
```
Comment (HLeithner) [inline libraries/src/Form/FormField.php]: Uppercase

File: tests/Unit/Libraries/Cms/Object/CMSObjectTest.php [hunk 10]
```diff
@@ -94,14 +94,14 @@ public function testGetProperties()
         $object = new CMSObject([
             '_privateproperty1' => 'valuep1',
             'property1'         => 'value1',
-            'property2'         => 5]);
+            'property2'         => 5, ]);
```
Comment (HLeithner) [inline tests/Unit/Libraries/Cms/Object/CMSObjectTest.php]: Looks broken

File: tests/Unit/Libraries/Cms/Uri/UriTest.php [hunk 11]
```diff
@@ -153,11 +153,11 @@ public function testBuildQuery(): void
...
                 'price' => [
                     'from' => 5,
-                    'to' => 10,
+                    'to'   => 10,
                 ],
-                'name' => 'foo'
+                'name' => 'foo',
             ],
-            'v' => 45];
+            'v' => 45, ];
```
Comment (HLeithner) [inline tests/Unit/Libraries/Cms/Uri/UriTest.php]: looks odd

**Review by HLeithner — CHANGES_REQUESTED** [review]
I did a full code review on all files but only for  real changes not for linebreaks and spaces. The PR looks ok except the 10 mentioned lines.

File: administrator/components/com_config/src/Controller/ComponentController.php (hunk 2 above)
Comment (laoneo) [reply administrator/components/com_config/src/Controller/ComponentController.php]: Fixed with hardcoded no break comment in the cs config

File: administrator/components/com_content/src/Controller/ArticleController.php (hunk 3 above)
Comment (laoneo) [reply administrator/components/com_content/src/Controller/ArticleController.php]: Corrected it in last commit, but this was also not detected by the CS fixer like in the display controllers

File: administrator/components/com_users/src/Model/UsersModel.php (hunk 4 above)
Comment (laoneo) [reply administrator/components/com_users/src/Model/UsersModel.php]: See comment above

File: components/com_contact/src/Controller/DisplayController.php (hunk 5 above)
Comment (laoneo) [reply components/com_contact/src/Controller/DisplayController.php]: Yes I know, but both fixers are accepting it

File: components/com_newsfeeds/src/Controller/DisplayController.php (hunk 6 above)
Comment (laoneo) [reply components/com_newsfeeds/src/Controller/DisplayController.php]: Same as above

File: libraries/src/Form/Field/TransitionField.php (hunk 7 above)
Comment (laoneo) [reply libraries/src/Form/Field/TransitionField.php]: It's correct, it is on multiple lines, but fixed it manually

File: libraries/src/Form/FormField.php (hunk 8 above)
Comment (laoneo) [reply libraries/src/Form/FormField.php]: See comment above

File: libraries/src/Form/FormField.php (hunk 9 above)
Comment (laoneo) [reply libraries/src/Form/FormField.php]: See comment above

File: tests/Unit/Libraries/Cms/Object/CMSObjectTest.php (hunk 10 above)
Comment (laoneo) [reply tests/Unit/Libraries/Cms/Object/CMSObjectTest.php]: Put the closing on a new line

File: tests/Unit/Libraries/Cms/Uri/UriTest.php (hunk 11 above)
Comment (laoneo) [reply tests/Unit/Libraries/Cms/Uri/UriTest.php]: Same as above

**Pull request merged**